src/server/plugins/engine/services/formsService.js

```diff
@@ -0,0 +1,37 @@
+import { formMetadataSchema } from '@defra/forms-model'
+
+import config from '~/src/server/config.js'
+import { getJson } from '~/src/server/plugins/engine/services/httpService.js'
+
+const { managerUrl } = config
+
+/**
+ * Retrieves a form definition from the form manager for a given slug
+ * @param {string} slug - the slug of the form
+ */
+export async function getFormMetadata(slug) {
+  const metadata = await getJson(`${managerUrl}/forms/slug/${slug}`)
+
+  // Run it through the schema to coerce dates
+  const result = formMetadataSchema.validate(metadata)
+
+  if (result.error) {
+    throw result.error
+  }
+
+  return result.value
+}
+
+/**
+ * Retrieves a form definition from the form manager for a given id
+ * @param {string} id - the id of the form
+ * @param {'draft'|'live'} state - the state of the form
+ */
+export async function getFormDefinition(id, state) {
+  const suffix = state === 'draft' ? '/draft' : ''
+  const defintion = await getJson(
+    `${managerUrl}/forms/${id}/definition${suffix}`
+  )
```

Comment:
I've fixed the types for the HTTP service in https://github.com/DEFRA/forms-runner/pull/95

It means you can now correctly type the payload, but it's a nested property unlike the plugin version of the HTTP service. Any good?

```suggestion
  const getJsonByType = /** @type {typeof getJson<FormDefinition>} */ (getJson)

  const suffix = state === 'draft' ? '/draft' : ''
  const { payload: definition } = await getJsonByType(
    `${managerUrl}/forms/${id}/definition${suffix}`
  )
```

Reply:
Here's a [branch](https://github.com/DEFRA/forms-runner/compare/feature/369420-run-forms-from-db...feature/369420-run-forms-from-db-rebased) with it all working in commit https://github.com/DEFRA/forms-runner/commit/851abff36eecb0f5f005107e886d9e064658daf8

Fixes all the ESLint errors in **formsService.js**

Reply:
Thanks Colin.

I will take this branch in once I've got this PR merged.
I was a little nervous doing it because of the other services currently using `src/server/services/httpService.ts`. i'm sure it's all fine but would like to do some testin/auditing first.